inc/saiswitch.h

```diff
@@ -605,41 +626,39 @@ typedef struct _sai_switch_notification_t
 
 /**
 * Routine Description:
-*   SDK initialization. After the call the capability attributes should be
+*   Create Switch 
+*   - SDK initialization. After the call the capability attributes should be
 *   ready for retrieval via sai_get_switch_attribute().
 *
 * Arguments:
-*   @param[in] profile_id - Handle for the switch profile.
-*   @param[in] switch_hardware_id - Switch hardware ID to open
-*   @param[in] firmware_path_name - Vendor specific path name of the firmware
-*                                   to load
-*   @param[in] switch_notifications - switch notification table
+*   @param[out] switch_id - router interface id
+*   @param[in] attr_count - number of attributes
+*   @param[in] attr_list - array of attributes
+*   @param[in] switch_notifications - switch notification table 
 * Return Values:
 *   @return SAI_STATUS_SUCCESS on success
 *           Failure status code on error
 */
-typedef sai_status_t (*sai_initialize_switch_fn)(
-    _In_ sai_switch_profile_id_t profile_id,
-    _In_reads_z_(SAI_MAX_HARDWARE_ID_LEN) char* switch_hardware_id,
-    _In_reads_opt_z_(SAI_MAX_FIRMWARE_PATH_NAME_LEN) char* firmware_path_name,
-    _In_ sai_switch_notification_t* switch_notifications
-    );
+typedef sai_status_t(*sai_create_switch_fn)(
+        _Out_ sai_object_id_t* switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list,
+        _In_ sai_switch_notification_t* switch_notifications
```

Comment:
for the switch notifications, can we also pass them as an attribute? In this case, it is function pointer?

Also, for all the notifications, we probably needs to add switch_id as an argument so the application knows which switch asic the notification is related to?